irrd/utils/text.py

```diff
@@ -97,3 +103,46 @@ def _str_to_camel_case(snake_str: str):
 
 def clean_ip_value_error(value_error):
     return re.sub(re_clean_ip_error, "", str(value_error))
+
+
+def dummify_object_text(rpsl_text: str, object_class: str, source: str, pk: str):
+    """
+    Modify the value of attributes in an RPSL object.
+    """
+
+    if not rpsl_text:
+        return rpsl_text
+
+    nrtm_response_dummy_object_class = get_nrtm_response_dummy_object_class_for_source(source)
+    if nrtm_response_dummy_object_class:
+        if object_class in nrtm_response_dummy_object_class:
+            dummy_attributes = get_setting(f"sources.{source}.nrtm_response_dummy_attributes")
+            if dummy_attributes:
+                if get_setting(f"sources.{source}.nrtm_response_dummy_remarks"):
+                    dummy_remarks = textwrap.indent(
+                        get_setting(f"sources.{source}.nrtm_response_dummy_remarks"), "remarks:".ljust(16)
+                    )
+                else:
+                    dummy_remarks = None
+
+                lines = rpsl_text.splitlines()
+
+                for index, line in enumerate(lines):
+                    for key, value in dummy_attributes.items():
+                        if "%s" in str(value):
+                            value = str(value).replace("%s", pk)
+
+                        if line.startswith(f"{key}:"):
+                            format_key = f"{key}:".ljust(RPSL_ATTRIBUTE_TEXT_WIDTH)
+                            if not isinstance(value, str):
+                                value = str(value)
+                            lines[index] = format_key + value
+
+                dummyfied_rpsl_object = "\n".join(lines) + "\n"
+
+                if rpsl_text != dummyfied_rpsl_object:
```

Comment:
Creating dummy_remarks could probably happen inside this if statement. So we don't calculate it if there is no change.

Reply:
This is how the current codes do. If no change or no remarks configured, we won't add the dummy remarks. Am I misunderstanding something here?